Aplib.Core/Desire/Goal.cs

```diff
@@ -0,0 +1,105 @@
+namespace Aplib.Core.Desire
+{
+    /// <summary>
+    /// A goal effectively combines a heuristicFunction with a tactic, and aims to meet the heuristicFunction by
```

Comment:
Let's try to leave as much implementation details such as variable names out of the summary:

```suggestion
    /// A goal effectively combines a heuristic function with a tactic, and aims to meet the heuristic function by
```

Reply:
Agreed!